crates/aranya-client/src/config/team.rs

```diff
@@ -13,14 +13,43 @@
 
 use aranya_daemon_api::TeamId;
 
-use crate::{error::InvalidArg, ConfigError, Result};
+use crate::{
+    config::quic_sync::{versioned::MaybeQuicSyncTeamInfo, AddTeamQuicSyncConfigBuilder},
+    error::InvalidArg,
+    ConfigError, Result,
+};
 
 pub mod quic_sync;
 pub use quic_sync::{
     AddTeamQuicSyncConfig, CreateTeamQuicSyncConfig, CreateTeamQuicSyncConfigBuilder,
 };
 
-/// Builder for [`CreateTeamConfig`].
+pub mod versioned {
+    #![allow(missing_docs)] // TODO: Document the type alias by generated by _obake_
+    #![allow(unused_macros)]
+    //! This module contains the [`TeamInfo`] type that is used to
+    //! initialize an [`super::AddTeamConfigBuilder`].
+
+    use serde::{Deserialize, Serialize};
+
+    use super::{quic_sync, TeamId};
+
+    // The doc comment below does not get applied to the generated type alias :(
+    /// Team data for initializing an [`super::AddTeamConfigBuilder`].
+    #[obake::versioned]
+    #[obake(version("0.1.0"))]
+    #[obake(derive(Clone, Debug, Serialize, Deserialize,))]
```

Comment:
Is there an extra comma at the end?
```suggestion
    #[obake(derive(Clone, Debug, Serialize, Deserialize))]
```